### PR TITLE
adding support for backing up and restoring KeyVault secrets

### PR DESCRIPTION
**Pull request by dragav**
adding support for backing up and restoring of KeyVault secrets

revising KeyVault backup/restore cmdlets:
    adding support for pipelining
    allowing overwriting of dest file
    using session data store for file operations

### CODE REVIEW

**Comment by cormacpayne** [comment]
Old PR: https://github.com/Azure/azure-powershell/pull/3787

**Comment by cormacpayne** [comment]
@azuresdkci add to whitelist

File: src/ResourceManager/KeyVault/Commands.KeyVault/Commands/BackupAzureKeyVaultSecret.cs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+                    ValueFromPipelineByPropertyName = true,
+                    ParameterSetName = BySecretNameParameterSet,
+                    HelpMessage = "Output file. The output file to store the backed up secret blob in. If not present, a default filename is chosen." )]
+        [Parameter( Mandatory = false,
+                    Position = 2,
+                    ValueFromPipelineByPropertyName = true,
+                    ParameterSetName = BySecretObjectParameterSet,
+                    HelpMessage = "Output file. The output file to store the backed up secret blob in. If not present, a default filename is chosen." )]
+        [ValidateNotNullOrEmpty]
+        public string OutputFile { get; set; }
```
Comment (dragav) [inline src/ResourceManager/KeyVault/Commands.KeyVault/Commands/BackupAzureKeyVaultSecret.cs]: Thanks; addressed in a subsequent commit, should show up shortly.

File: src/ResourceManager/KeyVault/Commands.KeyVault/Commands.KeyVault.csproj [hunk 2]
```diff
@@ -253,6 +255,10 @@
       <Project>{3819d8a7-c62c-4c47-8ddd-0332d9ce1252}</Project>
       <Name>Commands.ResourceManager.Common</Name>
     </ProjectReference>
+    <ProjectReference Include="..\..\Profile\Commands.Profile\Commands.Profile.csproj">
```
Comment (markcowl) [inline src/ResourceManager/KeyVault/Commands.KeyVault/Commands.KeyVault.csproj]: Please remove the reference to Commands.Profile, this is discouraged in cmdlet projects

**Comment by markcowl** [comment]
@dragav There are merge conflicts, please update

**Comment by markcowl** [comment]
@dragav There are merge conflicts, please update

File: src/ResourceManager/KeyVault/Commands.KeyVault/Commands.KeyVault.csproj (hunk 2 above)
Comment (dragav) [inline src/ResourceManager/KeyVault/Commands.KeyVault/Commands.KeyVault.csproj]: I think I added this reference so as not to duplicate the FileOverwrite messages. That couldn't be avoided, though. Reverted.

**Comment by dragav** [comment]
I picked up a bunch of commits following a faulty merge. Resubmitting a clean PR.

**Pull request closed without merging**